kasa/iot/iotdevice.py

```diff
@@ -679,7 +679,23 @@ async def wifi_join(self, ssid: str, password: str, keytype: str = "3") -> dict:
         async def _join(target: str, payload: dict) -> dict:
             return await self._query_helper(target, "set_stainfo", payload)
 
-        payload = {"ssid": ssid, "password": password, "key_type": int(keytype)}
+        try:
+            keytype = int(keytype)
+        except ValueError:
+            match keytype.lower():
```

Comment:
I think we should re-think this another way around, ie., by defining an enum for the key type for the common interface, and interpret that in *device implementations?

This is what executing `kasa wifi scan` reports on a couple of APs having different settings around me:
```
         WifiNetwork(ssid='TP-Link_xx', key_type='wpa2_psk', cipher_type=2, bssid='202xx', channel=10, rssi=None, signal_level=1)
         WifiNetwork(ssid='TP-Link_Guest_xx', key_type='none', cipher_type=0, bssid='AE1xx', channel=4, rssi=None, signal_level=1)
```

So for tapo devices, `keytype="open"` should be interpreted as `key_type='none', cipher_type=0`, and for `iotdevice` this would be only `key_type=0`.

What do you think?

Reply:
I can give it a shot. One of my major blindspots with this PR was not having devices that would respond to `kasa wifi scan`, so having that output helps quite a bit.

I'll have to research more about enums, but I'm happy to put a second attempt together, especially if it makes the code more resilient.

Reply:
I think this list from @aborno is probably the way to go from an Enum perspective. The `smartcam` devices appear to use the same numbering system so we can probably use an IntEnum here.

0 - Open
1 - WEP
2 - WPA
3 - WPA2
4 - WPA3

N.B. smartcam output:
```json
{"ap_list": [
    {"auth": 3,
    "bssid": "000000000000",
    "encryption": 2,
    "rssi": 2,
    "ssid": "I01BU0tFRF9TU0lEIw=="},
   {"auth": 0,
    "bssid": "000000000000",
    "encryption": 0,
    "rssi": 1,
    "ssid": "I01BU0tFRF9TU0lEIw=="}]
"wpa3_supported": "false"
}
```

Reply:
Some more examples here.
A SMART dev (P110): https://github.com/python-kasa/python-kasa/blob/223f3318ea81fd143dc53c1b94efbcd8e2565012/tests/fixtures/smart/P110(EU)_1.0_1.0.7.json#L462-L471 

For IOT (KP115):
```
{
  "netif": {
    "get_scaninfo": {
      "ap_list": [
        {
          "key_type": 3,
          "ssid": "#MASKED_SSID#"
        },
        {
          "key_type": 3,
          "ssid": "#MASKED_SSID#"
        }
      ],
      "err_code": 0,
      "wpa3_support": 1
    }
  }
}

```